File: packages/js-sdk/src/regex/detectors/sk/ico.ts
// Slovak ICO (company identification number) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { skIcoChecksum } from '../../validators'

class SkIcoDetector extends RegexDetector {
  entityType = EntityType.SK_ICO
  score = 0.85
  contextKeywords = ['ico', 'ičo', 'identifikacne cislo', 'identifikačné číslo', 'company id', 'business id']
  contextRequired = true

  pattern = /\b\d{8}\b/g
  validator = skIcoChecksum
}

registerRegion('sk', SkIcoDetector)

export { SkIcoDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/sk/ico.py
"""Slovak ICO (company identification number) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import sk_ico_checksum


@register_region("sk")
class IcoDetector(RegexDetector):
    entity_type = EntityType.SK_ICO
    score = 0.85
    context_keywords = [
        "ico", "\u00ed\u010do", "identifikacne cislo",
        "identifika\u010dn\u00e9 \u010d\u00edslo", "company id", "business id",
    ]
    context_required = True
    context_window = 50

    # 8-digit company ID
    pattern = re.compile(
        r"\b\d{8}\b"
    )
    validator = staticmethod(sk_ico_checksum)
